File: SRPTests/TestScripts/RenderTarget.py
# Test for creating custom render targets.

from SRPScripting import *
import utils

rt = ri.CreateRenderTarget()

testTexCallback = utils.GetTestTextureCallback(ri, rt, "FullscreenTexture_PS", "tex")

def RenderFrame(context):
	context.Clear((1, 0.5, 0, 1), [rt])
	testTexCallback()
	
ri.SetFrameCallback(RenderFrame)

File: SRPTests/TestScripts/CS/RenderTarget.cs
// Test for creating custom render targets.
#load "Utils/utils.cs"
using System.Numerics;

var rt = ri.CreateRenderTarget();

var testTexCallback = GetTestTextureCallback(rt, "FullscreenTexture_PS", "tex");

ri.SetFrameCallback(context =>
{
	context.Clear(new Vector4(1, 0.5f, 0, 1), new[] {rt});
	testTexCallback(context);
});
